### PR TITLE
fix: Don't read image from disk, when proxy is active

### PR DESCRIPTION
**Pull request by pat-richter**
When the proxy is enabled and getImageSize() can't find the file on disk, an PHP error is thrown.

It's an extension of the implementaion in productmain.tpl (look for the same comment)

### CODE REVIEW

**Comment by tmloberon** [comment]
yea!

**Comment by Sieg** [comment]
Hello @pat-richter and @tmloberon! I need some more input from your side, please, describe how to reproduce the problem, or link the related bug, please. In my case, it works just fine with varnish enabled and altimgurl configured. Do I need to remove my original shop images and have them on another server? What is your case?

If there is no bug created for this entry, would be great if you will report one, and describe reproduction steps there.

**Comment by pat-richter** [comment]
Hi @Sieg 

we're working on a local machine with docker. Since there are 20 GB of images or so, they are resolved trough a proxy from the live environment.

'HELP_ALTERNATIVE_IMAGE_SERVER_NOTE'
'Alternative URL to remote image server can be specified in configuration file config.inc.php by setting <i>sAltImageUrl</i> and <i>sSSLAltImageUrl</i>.<br> Thus all product pictures will be loaded from this alternative server instead of the local one. However, uploaded files will be stored locally. In this case synchronization to external server has to be done manually or with custom scripts.',

Kind regards,
Pat

File: tpl/page/details/inc/morepics.tpl [hunk 1]
```diff
@@ -11,8 +11,14 @@
         <ul class="[{if $iMorePics > 4}]slides[{else}]list-inline[{/if}]">
             [{oxscript add="var aMorePic=new Array();"}]
             [{foreach from=$oView->getIcons() key="iPicNr" item="oArtIcon" name="sMorePics"}]
-                [{assign var="sPictureName" value=$oPictureProduct->getPictureFieldValue("oxpic", $iPicNr)}]
-                [{assign var="aPictureInfo" value=$oConfig->getMasterPicturePath("product/`$iPicNr`/`$sPictureName`")|@getimagesize}]
+                [{* ToDo: This logical part should be ported into a core function. *}]
+                [{if $oConfig->getConfigParam('sAltImageUrl') || $oConfig->getConfigParam('sSSLAltImageUrl')}]
+                    [{assign var="aPictureInfo" value=$oPictureProduct->getMasterZoomPictureUrl($iPicNr)|@getimagesize}]
+                [{else}]
+                    [{assign var="sPictureName" value=$oPictureProduct->getPictureFieldValue("oxpic", $iPicNr)}]
```
Comment (Sieg) [inline tpl/page/details/inc/morepics.tpl]: This line should go out from the else, right?

**Comment by flow-control** [comment]
Hey @pat-richter,

I understand the problem, but don't like the desired solution.

Let me explain why:

For photoswipe to work it looks like we need to give the actual width and height of the image. To get this we currently fetch the path of the file and pass this to the `getimagesize()`.

With your desired change, the shop would fetch the image via a HTTP request to check the image size and then throw it away. Imaging having 5 images on this page, this would request five images via HTTP to just get the image size. Now imaging 100ms per image loading time, this would add 500ms to your overall TTFB. Nothing we want.

I get your point and understand your problem, clearly something we need to invest more time. Also as you stated in the comment, this calculations do not belong to the presentation layer.

I will talk to the PO about this problem.

/Flo

**Pull request closed without merging**